### PR TITLE
[539677] getValue should return null for null arg

### PR DESCRIPTION
**Pull request by kthoms**
Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=539677

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=539677

**Comment by cdietrich** [comment]
@szarnekow am not sure if this change is a good idea since it may cover user code errors. what do you think

**Comment by kthoms** [comment]
All clients in the Xtext code base are safe to retrieve null. Clients would have to catch the NPE that they would have experienced. I consider this as improvement. But good to have other viewpoints.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/SimpleAttributeResolver.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public T apply(EObject param) {
 	}
 
 	public T getValue(final K object) {
-		return valueCache.get(object);
+		return object != null ? valueCache.get(object) : null;
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/SimpleAttributeResolver.java]: Why was null passed in here at all?

**Pull request merged**